csa-runner-app/src/main/java/com/farao_community/farao/swe_csa/app/dichotomy/DichotomyRunner.java

```diff
@@ -170,14 +173,20 @@ public RaoResult runDichotomy(CsaRequest csaRequest) throws GlskLimitationExcept
                 LOGGER.info("Dichotomy stop criterion reached, CT PT-ES: {}, CT FR-ES: {}", Math.round(index.getBestValidDichotomyStepResult().getCounterTradingValues().getPtEsCt()), Math.round(index.getBestValidDichotomyStepResult().getCounterTradingValues().getFrEsCt()));
                 RaoResult raoResult = index.getBestValidDichotomyStepResult().getRaoResult();
 
-                RaoResultWithCounterTradeRangeActions raoResultWithRangeAction = getRaoResultWithCounterTradeRangeActions(network, crac, index, raoResult);
+                raoResult = updateRaoResultWithVoltageMonitoring(network, crac, raoResult, raoParameters);
+                RaoResultWithCounterTradeRangeActions raoResultWithRangeAction = updateRaoResultWithCounterTradingRAs(network, crac, index, raoResult);
                 fileExporter.saveRaoResultInArtifact(raoResultWithRangeAction, crac, Unit.AMPERE, csaRequest.getBusinessTimestamp());
                 return raoResultWithRangeAction;
             }
         }
     }
 
-    private RaoResultWithCounterTradeRangeActions getRaoResultWithCounterTradeRangeActions(Network network, Crac crac, Index index, RaoResult raoResult) {
+    private static RaoResult updateRaoResultWithVoltageMonitoring(Network network, Crac crac, RaoResult raoResult, RaoParameters raoParameters) {
+        VoltageMonitoring voltageMonitoring = new VoltageMonitoring(crac, network, raoResult);
+        return voltageMonitoring.runAndUpdateRaoResult(raoParameters.getLoadFlowAndSensitivityParameters().getLoadFlowProvider(), raoParameters.getLoadFlowAndSensitivityParameters().getSensitivityWithLoadFlowParameters().getLoadFlowParameters(), 4);
```

Comment:
it's not optimal to hard code 4, use `Runtime.getRuntime().availableProcessors()` instead ?